classes/SnippetLoader.php

```diff
@@ -0,0 +1,181 @@
+<?php namespace StudioAzura\W2C\Classes;
+
+use ApplicationException;
+use Cache;
+use Carbon\Carbon;
+use Cms\Classes\CmsException;
+use Cms\Classes\Controller as CmsController;
+use Cms\Classes\Page as CmsPage;
+use Cms\Classes\Theme;
+use Cms\Classes\ComponentManager;
+use Session;
+use SystemException;
+
+use Winter\Pages\Classes\SnippetManager;
+
+class SnippetLoader
+{
+    protected static $pageSnippetsCache = [];
+
+    /**
+     * Add a component registered as a snippet to the active controller.
+     *
+     * @param array $snippetInfo        The info of the snippet to register
+     * @return string                   The generated unique alias for this snippet
+     * @throws SystemException
+     * @throws CmsException
+     */
+    public static function registerComponentSnippet($snippetInfo)
+    {
+        $controller = CmsController::getController();
+
+        // Make an unique alias for this snippet based on its name and parameters
+        #$snippetInfo['code'] = uniqid($snippetInfo['code'] . '_' . md5(serialize($snippetInfo['properties'])) . '_');
+        // the line above was commented out to allow the overriden partials in theme to be used for the component alias
+
+        self::attachComponentSnippetToController($snippetInfo, $controller, true);
+        self::cacheSnippet($snippetInfo['code'], $snippetInfo);
```

Comment:
We prefer `static` over `self`, as `static` refers to the extended class if this class is extended upon. Could you also change this for all other `self` references in this file.

```suggestion
        static::attachComponentSnippetToController($snippetInfo, $controller, true);
        static::cacheSnippet($snippetInfo['code'], $snippetInfo);
```